### PR TITLE
Armor knowledge

### PR DESCRIPTION
**Pull request by ShinyPorygon**
All the armor pages, until you add more stuff.  Merry Christmas?

### CODE REVIEW

File: src/main/resources/assets/tconstruct/book/fantastic_foundry/index.json [hunk 1]
```diff
@@ -35,6 +35,20 @@
       }
     }
   },
+  {
+    "name": "plate",
```
Comment (KnightMiner) [inline src/main/resources/assets/tconstruct/book/fantastic_foundry/index.json]: If you name the section `armor`, you get an index automatically as that is the name I registered it under. Same for Materials and You.

Info would be a great "subtext" to the index section

File: src/main/resources/assets/tconstruct/book/mighty_smelting/en_us/abilities/armor/slurping.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
+{
+  "modifier_id": "tconstruct:slurping",
+  "text": [
+    {
+      "text": "Allows drinking fluid containing in the helmet to perform effects using the helmet keybind, mirroring effects from spilling."
```
Comment (KnightMiner) [inline src/main/resources/assets/tconstruct/book/mighty_smelting/en_us/abilities/armor/slurping.json]: This looks like the encyclopedia description

File: src/main/resources/assets/tconstruct/book/mighty_smelting/en_us/defense/protection/fire.json [hunk 3]
```diff
@@ -0,0 +1,14 @@
+{
+  "modifier_id": "tconstruct:fire_protection",
+  "text": [
+    {
+      "text": "Through the fire and the flames"
```
Comment (KnightMiner) [inline src/main/resources/assets/tconstruct/book/mighty_smelting/en_us/defense/protection/fire.json]: Missing a period at the end. Minor, I know, but darkosto will notice

File: src/main/resources/assets/tconstruct/book/mighty_smelting/en_us/defense/protection/turtle_shell.json [hunk 4]
```diff
@@ -0,0 +1,14 @@
+{
+  "modifier_id": "tconstruct:turtle_shell",
+  "text": [
+    {
+      "text": "Does not protect you from green, red, or blue shells."
+    }
+  ],
+  "more_text_space": true,
+  "effects": [
+    "Decreases damage underwater and increases wim speed",
```
Comment (KnightMiner) [inline src/main/resources/assets/tconstruct/book/mighty_smelting/en_us/defense/protection/turtle_shell.json]: I to enjoying wimming

File: src/main/resources/assets/tconstruct/book/mighty_smelting/en_us/defense/special/revitalizing.json [hunk 5]
```diff
@@ -0,0 +1,15 @@
+{
+  "modifier_id": "tconstruct:revitalizing",
+  "text": [
+    {
+      "text": "Jewled apples, what a throwback."
```
Comment (KnightMiner) [inline src/main/resources/assets/tconstruct/book/mighty_smelting/en_us/defense/special/revitalizing.json]: Am I missing a joke? I think you want `Jeweled`

File: src/main/resources/assets/tconstruct/book/puny_smelting/sections/upgrades.json [hunk 6]
```diff
@@ -119,5 +119,48 @@
...
+    "type": "modifier",
+    "data": "upgrades/armor/speedy.json"
+  },
+  {
+    "type": "modifier",
+    "data": "upgrades/armor/springy.json"
+  },
+  {
+    "type": "modifier",
+    "data": "upgrades/armor/sticky.json"
```
Comment (KnightMiner) [inline src/main/resources/assets/tconstruct/book/puny_smelting/sections/upgrades.json]: Should also be added to melee. I was lazy in the encyclopedia and just added the same page twice

**Comment by KnightMiner** [comment]
image

Also update this page in materials and you, since Thruul's plans are complete now

In addition, haste, experienced, and possibly another all have armor effects now, make sure you update those pages/add a second page for armor.

Also, luck has a different recipe behavior on leggings (max of 1 level), worth mentioning

File: src/main/resources/assets/tconstruct/book/mighty_smelting/en_us/abilities/armor/looting.json [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+  "text": [
+    {
+      "text": "Punching so hard that more comes out."
+    }
+  ],
+  "more_text_space": true,
+  "effects": [
+    "Increases mob drops and block drops for unarmed attacks",
+    "Maximum of 3 levels, different recipe for each level",
+    "Requires 1 Ability Slot for the first level, following levels do not need another slot"
```
Comment (KnightMiner) [inline src/main/resources/assets/tconstruct/book/mighty_smelting/en_us/abilities/armor/looting.json:13]: image
Little bit long, you probably want `more_text_space` to be false.

File: src/main/resources/assets/tconstruct/book/materials_and_you/en_us/modifiers/defense.json [hunk 8]
```diff
@@ -7,10 +7,6 @@
     {
       "text": "Like many other modifier types, defense modifiers require defense slots to be applied. Typically, armor with a large number of defense slots start with fewer upgrade slots, meaning functionality is sacrificed for protection.",
       "paragraph": true
-    },
-    {
-      "text": "The Encyclopedia of Tinkering is the best source on defense modifiers, though Thruul plans to include them in later editions of Mighty Smelting.",
```
Comment (KnightMiner) [inline src/main/resources/assets/tconstruct/book/materials_and_you/en_us/modifiers/defense.json:12]: I did not say delete this, I said update this. See the other modifier pages for examples, namely abilities. Goal is to tell the user which book(s) to read for modifier listings

**Comment by BeansBeefBroccoli** [comment]
> Merry Christmas?

Merry Christmas!

File: src/main/resources/assets/tconstruct/book/materials_and_you/en_us/modifiers/defense.json [hunk 9]
```diff
@@ -7,6 +7,10 @@
     {
       "text": "Like many other modifier types, defense modifiers require defense slots to be applied. Typically, armor with a large number of defense slots start with fewer upgrade slots, meaning functionality is sacrificed for protection.",
       "paragraph": true
+    },
+	 {
+      "text": "Abilities are beyond my knowledge but can be found in Mighty smelting, along with Defense modifiers.",
```
Comment (KnightMiner) [inline src/main/resources/assets/tconstruct/book/materials_and_you/en_us/modifiers/defense.json:12]: Let me clarify exactly what this should say. There are four pages in Materials and You:
* Upgrades: tells you to look in encyclopedia/puny
* Abilities: tells you to look in encyclopedia/mighty
* Slotless: tells you to look in encyclopedia/puny
* Defense: tells you to look in encyclopedia, and eventually mighty but not yet implemented

I want you to change defense to tell you defense modifiers are in encyclopedia and mighty smelting. You don't need to mention any other modifier types, the other pages cover those.

File: src/main/resources/assets/tconstruct/book/puny_smelting/en_us/upgrades/general.json [hunk 10]
```diff
@@ -3,6 +3,10 @@
   "text": [
     {
       "text": "Upgrades are modifiers added to tools using upgrade recipes. Each tool has a limited number of upgrade slots to start, though the number of slots can be increased in a few ways."
+    },
+	{
+      "text": "Many of the upgrades listed in general, weapon, bonus damage, and general work on armor as well!",
```
Comment (KnightMiner) [inline src/main/resources/assets/tconstruct/book/puny_smelting/en_us/upgrades/general.json:8]: Weapon/bonus damage only work on chestplates for the most part, which is tied to the unarmed modifier. I just had a special blurb in the encyclopedia for that. Let me know if you see any weapon modifiers that are not tied to unarmed attack.

Also said general twice.

**Pull request merged**